apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c

```diff
@@ -208,6 +219,51 @@ static int procfs_rewind_tc(const char *dirpath)
 
 	return OK;
 }
+#ifndef CONFIG_SMARTFS_MULTI_ROOT_DIRS
+void tc_fs_smartfs_mksmartfs(void)
+{
+	int ret;
+	ret = mksmartfs(SMARTFS_DEV_PATH, 1);
+	TC_ASSERT_EQ("mksmartfs", ret, OK);
+	ret = mksmartfs(INVALID_PATH, 1);
+	TC_ASSERT_NEQ("mksmartfs", ret, OK);
+	ret = mksmartfs(SMARTFS_DEV_PATH, 0);
+	TC_ASSERT_EQ("mksmartfs", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+#endif
+
+#ifndef CONFIG_FS_PROCFS_EXCLUDE_SMARTFS
+void tc_fs_smartfs_procfs_main(void)
+{
+	int fd;
+	int ret;
+	struct stat st;
+
+	fd = open(PROC_SMARTFS_PATH, O_RDONLY);
+	TC_ASSERT_GEQ("open", fd, 0);
+
+	ret = close(fd);
+	TC_ASSERT_EQ("close", ret, OK);
+
+	/* entry not found condition */
+
+	fd = open(PROC_SMARTFS_FILE_PATH, O_RDWR | O_CREAT );
+	TC_ASSERT_EQ("open", fd, ERROR);
+
+	fd = open(INVALID_PATH, O_RDONLY);
+	TC_ASSERT_EQ("open", fd, ERROR);
+
+	ret = read_dir_entries(PROC_SMARTFS_PATH);
+	TC_ASSERT_EQ("read_dir_entries", ret, OK);
+
+	ret = stat(PROC_SMARTFS_PATH, &st);
```

Comment:
@kavyasrinunna 
Please check if "st" also needs to be verified

Reply:
I think "st" may not be null, so verifying "st" may not be required.

Reply:
you can check st for not NULL using TC_ASSERT_NEQ